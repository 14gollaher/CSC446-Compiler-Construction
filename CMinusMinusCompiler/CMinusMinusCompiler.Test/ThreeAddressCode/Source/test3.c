int a,b,c,d;
int x,y;

int func(int a, int b)
{
  a = x/y + ( b-a);
  return 0;
}

int main()
{
   c=a+b*d;
   return 0;
}

