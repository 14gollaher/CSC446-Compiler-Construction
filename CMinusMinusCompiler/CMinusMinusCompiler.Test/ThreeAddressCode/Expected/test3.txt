PROC func
	_BP-2 = _x / _y
	_BP-6 = _BP+6 - _BP+4
	_BP-4 = _BP-2 + _BP-6
	_BP+4 = _BP-4
	_BP-8 = 0
	_AX = _BP-8
ENDP func

PROC main
	_BP-4 = _b * _d
	_BP-2 = _a + _BP-4
	_c = _BP-2
	_BP-6 = 0
	_AX = _BP-6
ENDP main

START PROC main

